lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -114,35 +148,33 @@ def get(self):
             # Ignore anything that doesn't use our API prefix, because it's
             # not in our API.
             if url.startswith(self.uri_prefix):
-                # If the URI is parameterized with a Flask "<type:name>"
-                # template string, we don't want to report it, so we remove
-                # it from the URI. We derive an API name by converting the
-                # "/" characters in the URI to "_", after removing a trailing
-                # "/" that would have been left by removing a template... we
-                # don't remove the trailing "/" from the URI, which serves as
-                # an indication that the parameter is needed.
-                #
-                # E.g, "/api/v1/controllers/list" yields:
-                #     "controllers_list": "/api/v1/controllers/list"
-                #
-                # while "/api/v1/users/<string:username>" yields:
-                #     "users": "/api/v1/users/"
-                #
-                # TODO: This won't work right with embedded template strings,
-                # which we're not currently using anywhere; but it'll require
-                # adjustment later if we add any. (E.g., something like
-                # "/api/v1/foo/<string:name>/detail/<string:param>")
+                simplified = self.param_template.sub(r"/{\g<name>}", url)
+                template: Dict[str, Any] = {"template": urljoin(host, simplified)}
+                params = []
+                for match in self.param_template.finditer(url):
+                    params.append(
+                        {"name": match.group("name"), "type": match.group("type")}
+                    )
+                template["params"] = params
                 url = self.param_template.sub("", url)
                 path = url[len(self.uri_prefix) + 1 :]
-                if path.endswith("/"):
-                    path = path[:-1]
                 path = path.replace("/", "_")
+
+                # Terminate the base URL with a slash if there were parameters
+                # removed. The client can just append a single parameter value.
+                # If there are two parameters, it'll have to add the separating
+                # "/" or use the template string: for example, with
+                # template.replace('{name1}', value1)
+                if params:
+                    url += "/"
```

Comment:
Is this of sufficient value to justify the complication?  That is, instead of doing this, why not just state that, if the client adds one or more parameters to the URI, it has to add the separating slashes, including for the first parameter?

Or, are we using the presence of the trailing slash for more than just making it easy to append the first parameter?

Reply:
I put that back in at the last minute for compatibility. It does have the small advantage of being able to do `endpoints.api.users + username` instead of worrying about the slash. I'd prefer to get rid of the `"api"` object entirely and just use the templated `"uri"` object, but I wanted the alternatives side by side to get feedback from @MVarshini @shubham-html-css-js.